src/behavior.ts

```diff
@@ -25,45 +25,63 @@
  */
 
 // d3
-import Selection = d3.Selection;
+import { BaseType, Selection } from "d3-selection";
 
 // powerbi.extensibility.utils.interactivity
-import { interactivityService } from "powerbi-visuals-utils-interactivityutils";
-import IInteractiveBehavior = interactivityService.IInteractiveBehavior;
-import ISelectionHandler = interactivityService.ISelectionHandler;
-import IInteractivityService = interactivityService.IInteractivityService;
+import { interactivityBaseService } from "powerbi-visuals-utils-interactivityutils";
+import IInteractiveBehavior = interactivityBaseService.IInteractiveBehavior;
+import ISelectionHandler = interactivityBaseService.ISelectionHandler;
+import IInteractivityService = interactivityBaseService.IInteractivityService;
 import { StreamGraphSeries } from "./dataInterfaces";
 import { getFillOpacity } from "./utils";
 
-export interface BehaviorOptions {
-    selection: Selection<d3.BaseType, StreamGraphSeries, any, any>;
-    clearCatcher: Selection<d3.BaseType, any, any, any>;
-    interactivityService: IInteractivityService;
+export interface BehaviorOptions extends interactivityBaseService.IBehaviorOptions<StreamGraphSeries>{
+    selection: Selection<BaseType, StreamGraphSeries, any, any>;
+    clearCatcher: Selection<BaseType, any, any, any>;
+    interactivityService: IInteractivityService<any>;
     series: StreamGraphSeries[];
 }
 
 const getEvent = () => require("d3-selection").event;
 
 export class StreamGraphBehavior implements IInteractiveBehavior {
-    private selection: Selection<d3.BaseType, StreamGraphSeries, any, any>;
-    private clearCatcher: Selection<d3.BaseType, any, any, any>;
-    private interactivityService: IInteractivityService;
+    private selection: Selection<BaseType, StreamGraphSeries, any, any>;
+    private clearCatcher: Selection<BaseType, any, any, any>;
+    private interactivityService: IInteractivityService<any>;
     private series: any = null;
 
+    protected options: BehaviorOptions;
+    protected selectionHandler: ISelectionHandler;
+
     public bindEvents(
         options: BehaviorOptions,
         selectionHandler: ISelectionHandler): void {
 
         this.selection = options.selection;
         this.clearCatcher = options.clearCatcher;
         this.interactivityService = options.interactivityService;
+        this.selectionHandler = selectionHandler;
 
         this.series = options.series;
 
-        this.selection.on("click", (series: StreamGraphSeries) => {
-            selectionHandler.handleSelection(
-                this.series[(<any>series).index],
-                (getEvent() as MouseEvent).ctrlKey);
+        this.selection.on("contextmenu", (datum) => {
```

Comment:
private handleContextMenu = () => {
        this.svg.on('contextmenu', (event: PointerEvent, dataPoint) => {
            this.selectionManager.showContextMenu(dataPoint ? dataPoint.identity : {}, {
                x: event.clientX,
                y: event.clientY
            });
            event.preventDefault();
        });
    }

Reply:
resolved